R/pmcmc.R

```diff
@@ -178,12 +164,38 @@ pmcmc_check_initial_nested <- function(initial, pars, n_chains) {
                      dimnames = list(nms, pops, NULL))
   }
 
-  ok <- apply(initial, 3, function(p) all(is.finite(pars$prior(p))))
+  pmcmc_check_initial_values(initial, pars, 3)
+}
+
+pmcmc_check_initial_values <- function(initial, pars, margin) {
```

Comment:
I think that margin here might be redundant as an arg - given the validation, is this always `length(dim(initial))`?